src/UniTask/Assets/Plugins/UniTask/Runtime/External/Localization/LocalizationAsyncExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿#if UNITASK_LOCALIZATION_SUPPORT
+
+using Cysharp.Threading.Tasks.Linq;
+using UnityEngine.Localization;
+
+namespace Cysharp.Threading.Tasks
+{
+    public static partial class LocalizationAsyncExtensions
+    {
+        public static IUniTaskAsyncEnumerable<string> OnValueChangedAsAsyncEnumerable(
+            this LocalizedString localizedString)
+        {
+            return UniTaskAsyncEnumerable.Create<string>(async (writer, cancellationToken) =>
+            {
+                async void Handler(string newValue)
+                {
+                    await writer.YieldAsync(newValue);
```

Comment:
What happens if the `writer.YieldAsync` is written twice before the consumer moved next? Does it queue the values? Does it explode?

Reply:
Even in the readme library there is an indication of this:

> A pull-type asynchronous stream does not get the next values until the asynchronous processing in the sequence is complete. This could spill data from push-type events such as buttons.

This is easily fixed with the help of a queue that has just the same Channel under the hood:

`LocalizedString.OnValueChangedAsAsyncEnumerable().Queue().Subscribe(Debug.Log);`